### PR TITLE
GITOPS-10535: add unit tests for core string and URL utilities

### PR DESCRIPTION
**Pull request by trdoyle81**
Adds unit tests for core git string/URL helpers (`stringHelpers`, `urls`) covering happy paths and edge cases (empty/malformed URLs, SHA detection, provider parsing).

Some updates: 
- Quiets PatternFly mock React warnings in ActionDropDown tests
- Enables `esModuleInterop` so `git-url-parse` default import works under Jest

Jira: 
https://redhat.atlassian.net/browse/GITOPS-10535

Note: 
Will need to be ported to release-4.18 also. 

Run tests: 
$ pnpm test 
$ pnpm test -- src/gitops/utils/stringHelpers.test.ts src/gitops/utils/urls.test.ts

### CODE REVIEW

File: src/gitops/utils/urls.ts [hunk 1]
```diff
@@ -3,7 +3,8 @@
  * https://github.com/argoproj/argo-cd/blob/4bd8b07c514e26c6b7837f30d52afd1a3cdedcfd/ui/src/app/shared/components/urls.ts
  */
 
-import * as GitUrlParse from 'git-url-parse';
+//cjs package — default import needs esmoduleinterop or jest can't call it
```
Comment (keithchong) [inline src/gitops/utils/urls.ts]: Thanks for catching and fixing this

**Review by keithchong — APPROVED** [review]

**Comment by keithchong** [comment]
@trdoyle81 , can you create another PR but apply this fix against the release-4.18 branch?
